bundles/org.dataflowanalysis.analysis/src/org/dataflowanalysis/analysis/core/AbstractVertex.java

```diff
@@ -0,0 +1,224 @@
+    }
+
+    /**
+     * Returns a list of characteristic literals that are set for a given characteristic type in the list of all node
+     * characteristics
+     * <p>
+     * See {@link AbstractVertex#getDataFlowCharacteristicNamesWithType(String)} for a similar method for data flow
+     * variables
+     * @param name Name of the characteristic type
+     * @return Returns a list of all characteristic literals matching the characteristic type
+     */
+    public List<String> getNodeCharacteristicNamesWithName(String name) {
+        return this.getAllNodeCharacteristics().stream().filter(cv -> cv.getTypeName().equals(name)).map(CharacteristicValue::getValueName)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns a list of characteristic literals that are set for a given characteristic type in the list of all node
+     * characteristics
+     * <p>
+     * See {@link AbstractVertex#getDataFlowCharacteristicIdsWithType(String)} for a similar method for data flow variables
+     * @param name Name of the characteristic type
+     * @return Returns a list of all characteristic literals matching the characteristic type
+     */
+    public List<String> getNodeCharacteristicIdsWithName(String name) {
+        return this.getAllNodeCharacteristics().stream().filter(cv -> cv.getTypeName().equals(name)).map(CharacteristicValue::getValueId)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns a List of IDs of characteristics and data flow variables that are set for a given characteristic type in the
+     * list of all data flow variables
+     * <p>
+     * See {@link AbstractVertex#getNodeCharacteristicIdsWithName(String)} for a similar method for node characteristics
+     * @param type Name of the characteristic type
+     * @return Returns a list of all characteristic literals matching the characteristic type
+     */
+    public List<List<String>> getDataFlowCharacteristicIdsWithType(String type) {
+        List<List<String>> dfCharIds = new ArrayList<>();
+        for (DataFlowVariable df : this.getAllDataFlowVariables()) {
+            List<String> charValueIds = new ArrayList<>();
+            for (CharacteristicValue charValue : df.getAllCharacteristics()) {
+                if (charValue.getTypeName().equals(type)) {
+                    charValueIds.add(charValue.getValueId());
+                }
+            }
+            dfCharIds.add(charValueIds);
+        }
+
+        return dfCharIds;
+    }
+
+    /**
+     * Returns a List of names of characteristics and dataflow variables that are set for a given characteristic type in the
+     * list of all data flow variables
+     * <p>
+     * See {@link AbstractVertex#getNodeCharacteristicNamesWithName(String)} for a similar method for node characteristics
+     * @param type Name of the characteristic type
+     * @return Returns a list of all characteristic literals matching the characteristic type
```

Comment:
Also returns a list of lists what do these methods actually do?

Reply:
See discussion here: https://github.com/DataFlowAnalysis/DataFlowAnalysis/pull/123#discussion_r1509950587 and in commit 8f74e1c836e0a214fc8c9e3f5ae0f2214ea2032d